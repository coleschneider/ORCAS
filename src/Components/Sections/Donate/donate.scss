

/* ===================================================================
 * # donate
 *
 * ------------------------------------------------------------------- */
 .s-donate {
  text-align: center;
  padding-top: 5rem;
  padding-bottom: 5rem;
  background-color: #eaeaea;
  position: relative;
}


/* ------------------------------------------------------------------- 
 * ## donate plan blocks
 * ------------------------------------------------------------------- */
.plans {
  margin-top: 6.4rem;
  max-width: 840px;
}

.item-plan {
  padding: 0 30px;
}

.item-plan__block {
  background: #ffffff;
  border-radius: 4px;
  padding: 5.6rem 4.2rem 5.4rem;
  margin-bottom: 4rem;
}

.item-plan__title {
  font-weight: 700;
  font-size: 2.1rem;
  line-height: 1.333;
  color: #000000;
  margin: 0 0 1.6rem 0;
}

.item-plan__price,
.item-plan__per {
  display: inline-block;
  margin-bottom: 0;
}

.item-plan__price {
  font-family: "Chivo", sans-serif;
  font-size: 7.4rem;
  line-height: 1.081;
  font-weight: 600;
  color: #000000;
}

.item-plan__per {
  font-size: 1.6rem;
  line-height: 1;
  color: rgba(0, 0, 0, 0.4);
  margin: 0 4px;
}

.item-plan__bottom-part {
  margin-top: 4.4rem;
}

.item-plan__services {
  font-size: 1.7rem;
  line-height: 1.882;
  margin-left: 0;
}

.item-plan__services li {
  display: list-item;
  list-style: none;
  padding: .2rem 0 .2rem .8rem;
  position: relative;
}

.item-plan__services li::before {
  content: "\f00c";
  font-family: 'Font Awesome 5 Free';
  font-size: 1.2rem;
  font-weight: 900;
  display: inline-block;
  color: #80d3d2;
  margin-right: 18px;
}

.item-plan--popular {
  color: white;
}

.item-plan--popular .item-plan__services li::before {
  color: #ffffff;
}

.item-plan--popular .item-plan__block {
  background-color: #80d3d2;
  border: none;
}

.item-plan--popular .item-plan__title,
.item-plan--popular .item-plan__price {
  color: #ffffff;
}

.item-plan--popular .item-plan__per {
  color: rgba(255, 255, 255, 0.7);
}

.item-plan--popular .btn {
  background-color: #ffffff;
  border: 0.2rem solid #ffffff;
  color: #000000;
}

.item-plan--popular .btn:hover,
.item-plan--popular .btn:focus {
  background-color: #433a6e;
  border: 0.2rem solid #433a6e;
  color: #ffffff;
}


/* ------------------------------------------------------------------- 
 * responsive:
 * donate
 * ------------------------------------------------------------------- */
@media only screen and (max-width:1200px) {
  .s-donate {
    padding-top: 18.4rem;
  }

  .plans {
    max-width: 800px;
  }

}

@media only screen and (max-width:1000px) {
  .s-donate {
    padding-top: 17.2rem;
    padding-bottom: 17.2rem;
  }

}

@media only screen and (max-width:800px) {
  .s-donate {
    padding-top: 16rem;
    padding-bottom: 14.8rem;
  }

  .plans {
    max-width: 500px;
  }

}

@media only screen and (max-width:600px) {
  .s-donate {
    padding-top: 14.8rem;
    padding-bottom: 12rem;
  }

}

@media only screen and (max-width:450px) {
  .item-plan {
    padding: 0 20px;
  }

}

@media only screen and (max-width:350px) {
  .item-plan__price {
    font-size: 5.6rem;
  }

  .item-plan__block {
    padding: 5.4rem 3.2rem 4.8rem;
  }

}