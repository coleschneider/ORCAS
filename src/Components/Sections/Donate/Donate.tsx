import * as React from 'react';
import ShortDonation from './ShortDonation/ShortDonation';
import './donate.scss';
import { createSection } from 'utils/sectionUtils';
interface DonateState {
  tryFormSubmission: boolean;
}

const DonateWrapper = createSection({
  description: `Et nihil atque ex. Reiciendis et rerum ut voluptate. Omnis molestiae nemo est. Ut quis enim rerum quia
  assumenda repudiandae non cumque qui. Amet repellat omnis ea.`,
  displayTitle: 'You Can Help with your Donation.',
  sectionName: 'donate',
});

class Donate extends React.Component<{}, DonateState> {
  submitDonation = null;
  handleSubmit = e => {
    if (this.submitDonation) {
      this.submitDonation(e);
    }
  };
  bindSubmitForm = (submitDonation: () => void) => {
    this.submitDonation = submitDonation;
  };
  render() {
    return (
      <DonateWrapper>
        <div className="row plans block-1-2 block-m-1-2 block-tab-full stack">
          <div className="col-block item-plan">
            <div className="item-plan__block">
              <div className="item-plan__top-part">
                <h3 className="item-plan__title">Give</h3>
                <p className="item-plan__price">$5</p>
                <p className="item-plan__per">To ORCAS</p>
              </div>
              <div className="item-plan__bottom-part">
                <ul className="item-plan__mission give-five">
                  <li>
                    <span>Email</span> Updates
                  </li>
                </ul>
                <a className="btn btn--primary large full-width" href="#0">
                  Give $5
                </a>
              </div>
            </div>
          </div>
          <div className="col-block item-plan item-plan--popular">
            <div className="item-plan__block">
              <ShortDonation bindSubmitForm={this.bindSubmitForm} />
              <div className="item-plan__bottom-part">
                <ul className="item-plan__mission">
                  <li>
                    <span>Give</span> Monthly
                  </li>
                  <li>
                    <span>Another's</span> Behalf
                  </li>
                  <li>
                    <span>Email</span> Updates
                  </li>
                  <li>Subscribe</li>
                </ul>
                <div className="btn btn--primary large full-width" onClick={this.handleSubmit}>
                  Donate
                </div>
              </div>
            </div>
          </div>
        </div>
      </DonateWrapper>
    );
  }
}

export default Donate;
