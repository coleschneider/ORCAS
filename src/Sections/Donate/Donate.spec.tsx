import * as React from 'react';
import Donate from './Donate';

describe('Donate', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Donate />);
    expect(wrapper).toMatchSnapshot();
  });
});
