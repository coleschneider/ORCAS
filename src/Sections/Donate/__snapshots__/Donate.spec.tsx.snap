// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donate matches the snapshot 1`] = `
<section
  className="s-donate target-section"
  id="donate"
>
  <div
    className="row section-header align-center"
  >
    <div
      className="col-full"
    >
      <h1
        className="display-1"
      >
        You Can Help 
        <br />
         with your Donation.
      </h1>
      <p
        className="lead"
      >
        Et nihil atque ex. Reiciendis et rerum ut voluptate. Omnis molestiae nemo est. Ut quis enim rerum quia assumenda repudiandae non cumque qui. Amet repellat omnis ea.
      </p>
    </div>
  </div>
  <div
    className="row plans block-1-2 block-m-1-2 block-tab-full stack"
  >
    <div
      className="col-block item-plan"
    >
      <div
        className="item-plan__block"
      >
        <div
          className="item-plan__top-part"
        >
          <h3
            className="item-plan__title"
          >
            Give
          </h3>
          <p
            className="item-plan__price"
          >
            $5
          </p>
          <p
            className="item-plan__per"
          >
            To ORCAS
          </p>
        </div>
        <div
          className="item-plan__bottom-part"
        >
          <ul
            className="item-plan__mission"
          >
            <li>
              <span>
                Email
              </span>
               Updates
            </li>
          </ul>
          <a
            className="btn btn--primary large full-width"
            href="#0"
          >
            Give $5
          </a>
        </div>
      </div>
    </div>
    <div
      className="col-block item-plan item-plan--popular"
    >
      <div
        className="item-plan__block"
      >
        <div
          className="item-plan__top-part"
        >
          <h3
            className="item-plan__title"
          >
            Another amount?
          </h3>
          <p
            className="item-plan__price"
          >
            <input
              placeholder="Enter an amount to give"
              type="text"
            />
          </p>
          <p
            className="item-plan__per"
          >
            Or give monthly
          </p>
        </div>
        <div
          className="item-plan__bottom-part"
        >
          <ul
            className="item-plan__mission"
          >
            <li>
              <span>
                Give
              </span>
               Monthly
            </li>
            <li>
              <span>
                Another's
              </span>
               Behalf
            </li>
            <li>
              <span>
                Email
              </span>
               Updates
            </li>
            <li>
              Subscribe
            </li>
          </ul>
          <a
            className="btn btn--primary large full-width"
            href="#0"
          >
            Donate
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
